### PR TITLE
reorganize ClickException in Trakt AuthException

### PR DESCRIPTION
**Pull request by twolaw**
Exception in Exception in Exception is too noisy

No Trakt connection -> stop the script with CRITICAL error

### CODE REVIEW

**Comment by glensc** [comment]
what is the problem you trying to solve? show some output before/after.

File: plextraktsync/trakt/TraktApi.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def me(self):
             return trakt.users.User("me")
         except (OAuthException, ForbiddenException) as e:
             logger.fatal(f"Trakt authentication error: {str(e)}")
-            raise ClickException(e)
+            sys.exit(1)
```
Comment (glensc) [inline plextraktsync/trakt/TraktApi.py:56]: sys.exit in a class? that's a no!

class methods should throw, and this is what it is already doing now.

Comment (twolaw) [reply plextraktsync/trakt/TraktApi.py:56]: Ok but at the moment we have 3 cascaded exceptions throwed if bad trakt authentication :
OAuthException > ClickException > TypeError

Isn't that bad practice ?

My first fix to remove TypeError was :
```diff
- raise ClickException(e)
+ raise ClickException(str(e))
```

Comment (glensc) [reply plextraktsync/trakt/TraktApi.py:56]: you should add also to cli.py:

```py
            except ClickException as e:
                raise ClickException(f"Error running {name} command: {e}")
```

before:
```py
            except Exception as e:
                from plextraktsync.factory import logger
                logger.exception(e)

```

so, that `ClickException` is exiting without printing the trace (too noisy).

**Comment by twolaw** [comment]
Fixed.
No more exceptions cascade when Trakt authentication fail.
Just a CRTITICAL error.

**Review by glensc — APPROVED** [review]

**Pull request merged**